src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,118 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> header;
+    private int size;
+
+    public MyLinkedList() {
+        header = new Node<>(null, null, null);
+        header.next = header;
+        header.prev = header;
+        size = 0;
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        add(value, size);
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        Node<T> entryPoint = index == size ? header : entry(index);
+        Node<T> element = new Node<>(value, entryPoint, entryPoint.prev);
+        element.prev.next = element;
+        element.next.prev = element;
+        size++;
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T element : list) {
+            add(element);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return entry(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> element = entry(index);
+        T prevValue = element.item;
+        element.item = value;
+        return prevValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> element = entry(index);
+        element.prev.next = element.next;
+        element.next.prev = element.prev;
+        element.next = null;
+        element.prev = null;
+        T value = element.item;
+        element.item = null;
+        size--;
```

Comment:
let's move this code to separate private method `unlink(Node<T> nodeToRemove)` and then we can use it in method remove(T t)